### ISSUE TITLE
Learn to Code Resources

### ISSUE THREAD

**Issue by leighphan** [codelaboc/hellocodelab#12]
List your favorite learn to code resources here. This will serve as future reference for our group via github/future website.

Examples include online courses, interactive resources.

**Comment by leighphan** [comment]
# Command Line + Set up

**Command Line the Hard Way**: http://cli.learncodethehardway.org/book/

**Command Line Bootcamp**: http://rik.smith-unna.com/command_line_bootcamp/?id=9kvibsqyr0xeescgx5eetrzfr

**Homebrew:** http://brew.sh/

# Intro-Level: Web Dev
- **Learn Code the Hard Way**: http://learncodethehardway.org/
  _Python, Ruby, C, SQL, Regex_
- **Codecademy**: https://www.codecademy.com/
  _Web Dev_
- **Code School**: https://www.codeschool.com/
  _HTML/CSS, JavaScript, Ruby, Python, .NET, iOS, Git, Database_
- **FreeCodeCamp**: https://freecodecamp.com/
- **Exercism**: http://exercism.io/

## JavaScript
- **Node School**: http://nodeschool.io/

**Comment by chengyus** [comment]
Rails Tutorial dot Org (Free Online Book): https://www.railstutorial.org/book

If you read Traditional Chinese, then please reference (or plus Google Translate my Ruby mentor's site, but unfortunately it's only up-to-date for Rails 4.2 I think, since Rails 5 is just recently released): https://ihower.tw/rails4/  (Ruby on Rails 實戰聖經)

And obviously the defacto site for Rails guide: (but unfortunately, I feel that site is just version controlled and not owned by one author, so sometimes you have to refer to the edge guide and some times for older versions the regular guide). http://guides.rubyonrails.org/  vs http://edgeguides.rubyonrails.org/   (but now that rails 5 is out, the differences should be little?)

Some stuff I got influenced thru training (ISC(2) CSSLP, not certified though, cuz I don't believe in paying certifications periodically): https://www.owasp.org/index.php/Secure_SDLC_Cheat_Sheet

Agile Kanban, why I use Trello, JIRA Agile Kanban (used to be called greenhopper), I just heard Blossom last meeting (https://www.blossom.co/kanban), nice :
https://en.wikipedia.org/wiki/Kanban_(development)

**Comment by thebearingedge** [comment]
JavaScript Fundamentals (For Beginners and Experienced Programmers) FTW!
[How To Learn JavaScript Properly](http://javascriptissexy.com/how-to-learn-javascript-properly/)

This guide and accompanying books helped me so much!

**Comment by IanDCarroll** [comment]
Some stuff compiled at FullStackLA: https://github.com/fullstackla/pairing-meetup/commit/8df8c630c5269d52e04f7d9c3b2ff11c4adbbf58

**Comment by dotproto** [comment]
### JavaScript & CSS Resources
- [Mozilla Developer Network (MDN)](https://developer.mozilla.org/en-US/) is a fantastic resource for  [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference), [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference)[(3)](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS3), and [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML)[(5)](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5). Basically this is your hardcore dictionary reference.
- [CSS Tricks](https://css-tricks.com) has a lot of articles on CSS and JavaScript. A lot of handy posts on specific technologies, like [flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)

### Web Platform Compatibility
- [Can I Use?](https://caniuse.com) is a great reference for checking browser support for various web platform features.
- [I Want to Use](http://www.iwanttouse.com) lets you add a bunch of web technologies to check how many users will be able to visit your site.
- Microsoft's [Platform Status](https://developer.microsoft.com/en-us/microsoft-edge/platform/status/) keeps track of new web feature implementation status across all major browsers.

**Comment by interdigitize** [comment]
JavaScript Resources:
- **You Don't Know JS**: https://github.com/getify/You-Dont-Know-JS
- **Functional Programing Practice — LearnRX**: http://reactivex.io/learnrx/

**Comment by stoilsky** [comment]
www.w3schools.com covers the most fundamental web stuff (html, css, js, bootstrap etc) in a good way for beginners

**Comment by dotproto** [comment]
Good point, @stoich. W3 Schools gets crap from professional developers, but it's a great resource when getting started.

**Comment by IanDCarroll** [comment]
Wouldn't it be cool if we took this hodgepodge of recommendations and organized them into tracks of curricula?

**Comment by leighphan** [comment]
This is intended to be a general resource for our website. For those interested, we're working on exercism.io next meeting!

**Comment by ntreadway** [comment]
Swift and Unity tutorials https://www.raywenderlich.com/

**Comment by kangashley** [comment]
@bluAsterisk @davidws22

## Reads on JavaScript, vanilla-flavored <3:

[JavaScript: The Good Parts](http://shop.oreilly.com/product/9780596517748.do)
[Eloquent JavaScript](http://eloquentjavascript.net/)
[You Don't Know JS](https://github.com/getify/You-Dont-Know-JS)
[Learning JavaScript Design Patterns](https://addyosmani.com/resources/essentialjsdesignpatterns/book/)
